ui/app/components/auth/form/base.ts

```diff
@@ -57,16 +57,16 @@ export default class AuthBase extends Component<Args> {
           selectedAuth: this.args.authType,
         });
 
-        this.handleAuthResponse(authResponse);
+        this.handleAuthResponse(authResponse, this.args.authType, formData);
       } catch (error) {
         this.onError(error as Error);
       }
     })
   );
 
-  handleAuthResponse(authResponse: AuthData) {
+  handleAuthResponse(authResponse: AuthData, authType: string, formData?: object) {
```

Comment:
passing to support mfa validation for standard methods, but this is going to check a bit when everything is wired up because the end goal is for mfa validation to be handled the same for all methods

Reply:
I'm guessing `formData` is different depending on the `authType`? It might be nice to eventually have it typed but for now `Record<string, unknown>` might be better.

Reply:
Agreed! I typed it explicitly in the "complex" scenarios: 
https://github.com/hashicorp/vault/blob/10f9cf2f294a9372b1c2ae6f6acd64f86ef336e2/ui/app/components/auth/form/oidc-jwt.ts#L31-L41

but this whole part will get a makeover when the auth service is cleaned up. Because everything is so abstracted so it can be passed to the cluster adapter's [authenticate](https://github.com/hashicorp/vault/blob/main/ui/app/adapters/cluster.js#L124) method, it's been a pain to figure out what data is actually needed where. 

But this will get cleared up when things are actually wired up and even more so when the api service is being used 🎉

Reply:
Nice sounds good!